lib/Array.rb

```diff
@@ -0,0 +1,14 @@
+class Array
+
+	def update_with(array)
+		
+		self.each do |element| 
+			array = yield array,element
+		end
+		p array
+	end
+
+
+end
+
+
```

Comment:
Indentation - change the preferences in Sublime to ensure you are using 2 spaces instead of tabs.